src/anemoi/models/layers/mapper.py

```diff
@@ -127,8 +127,12 @@ def _register_edges(self, sub_graph: HeteroData, src_size: int, dst_size: int, t
         trainable_size : int
             Trainable tensor size
         """
-        self.edge_dim = sub_graph.edge_attr.shape[1] + trainable_size
-        self.register_buffer("edge_attr", sub_graph.edge_attr, persistent=False)
+        edge_attrs = sub_graph.edge_attrs()
+        edge_attrs.remove("edge_index")
```

Comment:
Can we have a comment why we do this please?

Reply:
I would say so, but I also see this part a bit messy. Would be better if we assume for the moment that all mappers take the same edge attributes?